### PR TITLE
Fix message formater `HasLength::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Rule/HasLength.php [hunk 1]
```diff
@@ -39,11 +39,13 @@ final class HasLength extends Rule
     /**
      * @var string user-defined error message used when the length of the value is smaller than {@see $min}.
      */
-    private string $tooShortMessage = 'This value should contain at least {min, number} {min, plural, one{character} other{characters}}.';
```
Comment (samdark) [inline src/Rule/HasLength.php:42]: Is there a reason why `min` plural formatting was removed in this an other cases?

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: https://github.com/yiisoft/validator/blob/b2276a349191c00e9e829583712c546a10edf463/src/Formatter.php#L11 The formatter currently does not accept this format, getting the message without formatting.

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: image

Comment (xepozz) [reply src/Rule/HasLength.php:42]: How about to just improve Formatter instead?

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: > How about to just improve Formatter instead?

Yes it's a good idea, but the formatter should be general, it should be a separate package

Comment (samdark) [reply src/Rule/HasLength.php:42]: We should not remove plurals. It's better to implenent basic fallback for English language in https://github.com/yiisoft/validator/blob/master/src/Formatter.php.

Comment (rustamwin) [reply src/Rule/HasLength.php:42]: Use https://github.com/yiisoft/translator-formatter-intl/blob/master/src/IntlMessageFormatter.php instead.

Comment (samdark) [reply src/Rule/HasLength.php:42]: @rustamwin nope, can't do that because that would create a hard dependency.

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: @samdark  and how will we add support for validator translations ?

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: if the idea is that translation support is outside the package it should be added to the `README.md`.

Comment (rustamwin) [reply src/Rule/HasLength.php:42]: @samdark Then `MessageFormatter` from intl should be used in https://github.com/yiisoft/validator/blob/master/src/Formatter.php?

Comment (samdark) [reply src/Rule/HasLength.php:42]: Alternatively we can use https://github.com/yiisoft/translator/blob/master/src/MessageFormatterInterface.php and require it being specified. That's even better.

Comment (terabytesoftw) [reply src/Rule/HasLength.php:42]: > Alternatively we can use https://github.com/yiisoft/translator/blob/master/src/MessageFormatterInterface.php and require it being specified. That's even better.

And why not use the package in https://github.com/yiisoft/validator/blob/master/src/Formatter.php at the end will be the same dependencies?

Comment (samdark) [reply src/Rule/HasLength.php:42]: I am suggesting exactly that.

**Review by samdark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**